### DISCUSSION TITLE
Audiobookshelf app - user experience report after 6 weeks of intensive power usage. How is yours?

### DISCUSSION THREAD

**Discussion by sandreas** [advplyr/audiobookshelf-app#864] [Show and tell]
Hey @advplyr,

I thought you'd maybe interested in a detailed user experience report of a (not so common) power user - or maybe other users add their experiences to help you prioritize the right features. Over the last 6 weeks I switched from my good old iPod Nano 7 (yeah, I really use this device) over to the Audiobookshelf and tried to note the pros and cons.

@all Feel free to add your own report below, so that I maybe learn something.

## Introduction
First let me say that Audiobookshelf is awesome... it's free, it is selfhosted and comes with a native app, that pretty much works like expected. However, there are some issues... that's why I decided to write down this report.

## Hardware / Software
- [Unihertz Jelly 2e](https://www.unihertz.com/products/jelly-2e) - a mini smartphone dedicated for audio book playback
  - WITH audio jack, 512GB MicroSD support and customizable button 
- [Samsung EG920 EO-EG920BW](https://www.amazon.de/Samsung-EG920-EO-EG920BW-Kopfh%C3%B6rer-Galaxy-wei%C3%9F/dp/B00YSPG7J6) - cheap EarPod like headphones with remote
- Android 12 (Stock)
- Audiobookshelf app (latest)

## Hardware Alternatives
Here are some alternative players I tried or considered, while in the end I decided for the Unihertz Jelly 2e.

- **Fiio M6** (used) - pretty interesting android 7 player with headphone remote control support, unfortunately ABS did not work
- **Sony NW A306** - Android 12 Player with control buttons, very interesting, BUT remote control did not work with audio-jack. I did order a USB-C to Audio-Jack Adapter and it worked, but 400 bucks was just too much
- **Satrend S11** - Smartphone - Android 7 - no longer available in my country
- **LG G5 H850** - pretty bulky but powerful device, audio jack and 
## Use cases

I use Audiobookshelf mainly to listen to audio books (no podcasts). Here are some more specific use cases:

- Going for a walk with my 2 kids
- Putting my 1 1/2 year old to bed (either carrying him around or laying in bed besides him when its already dark)
- Driving in the car
- Cleaning up the kitchen on my own

In the first two cases it is really important to remote control the phone, either with the customizable media button in my pocket or even better via headphone remote. If the kids are sleeping and the room is dark, taking out the phone to restart the app or control the status is a no go, because the kids might wake up because of the bright light.

## Streaming vs offline mode

Since Audiobookshelf cannot be used via subfolder in a reverse proxy scenario (see [this issue](https://github.com/advplyr/audiobookshelf/issues/385)), I'm unfortunately forced to sync my books at home via wifi and then listen to them offline on the go. This is not my favoured scenario, but it's ok, since my iPod Nano had to be synced via cable, so I still consider this as an advantage.

## Most important features
The most important features I use:

- Continue books, Latest books and search for audio books
- Wifi sync / offline mode
- Headphone control (e.g. when I'm putting my children to bed, I cannot pull out my phone with a bright screen and select the track)
- Lockscreen controls (e.g. if I HAVE TO pull out the phone, I don't have to login but can play pause and rewind from locked screen)
- Media-Button (the Unihertz Jelly 2e has a custom programmable button, that I use to play/pause media

## Pros
- Progress saving for continue books works very good
- Wifi sync works ok
- Headphone remote control (kind of) works most of the time
- Lockscreen controls work flawlessly

## Cons
- App crashes pretty often while playing without any reproducible pattern (this is something really annoying because I have to take out the phone, unlock, restart the app, select the book again and press play - the bright screen is a risk to wake up the children)
  - https://github.com/advplyr/audiobookshelf-app/issues/843
  - https://github.com/advplyr/audiobookshelf-app/issues/773
  - https://github.com/advplyr/audiobookshelf-app/issues/608
- After app restart / crash it does not switch to the player with the last auidobook but the main menu and you have to select the last book again
- Startup and load times - on every start there is a *loading* indicator, even in offline mode - this is annoying after a crash, when I just want to resume the last book I listened to
- The headphone remote does not work reliably (there is no docs on which button codes are supposed to control which function, and they do not work reliably - play/pause works, but rewind, skip etc. don't work as expected most of the time)
  - see https://github.com/advplyr/audiobookshelf-app/issues/847
- On Wifi sync I have to download manually and always select the location where to download to (a direct download with a default  location would be better in my opinion)
  - see https://github.com/advplyr/audiobookshelf-app/discussions/846
  - and https://github.com/advplyr/audiobookshelf-app/issues/774

## iPod Nano 7 comparison

The iPod Nano 7 is an exceptional device, although it is an Apple device. Basically, they thought of a lot of things others did not and the fact that the device was released 11 (!) years ago, I cannot state enough how impressed I am.

### Sync
Obviously, the iPod Nano 7 cannot be synced over Wifi, so ABS is the clear winner here

### Library management
The iPod nano is limited to 16GB storage, so there is no search. However, the UI and organisation has its pros - for me ABS is the winner, but it was close.

### Efficiency / Size
The iPod is damn small and with a fresh battery and bluetooth disabled, it can play a long time. The battery is 220mah, so this is really impressive. This also makes the iPod significantly smaller than any other android device I found. iPod wins.

### Playback / Robustness
The iPod is nearly flawless... Playback just works and the device can be controlled via buttons (vol+, vol-, toggle, on/off, home). The EarPods remote does work exactly the same as the device buttons and they support the same button codes (see https://github.com/advplyr/audiobookshelf-app/issues/847). The only issue I had so far was after a longer time without usage I could not start playback with the headphone remote, but as soon as I pressed the on button, it started.
iPod is the clear winner here.
 
## Conclusion
ABS is not a commercial product and I'm glad, it exists. It's way better that most alternatives I tried. However, I think we can learn a lot from Apples devices here.

**Comment by brcarls** [comment]
Great review!

I think the app crashes more often when it doesn't have focus... My prime use is audiobooks when driving, so mapping always has focus (usually Waze).    The crashes were so frustrating and distracting that I don't use the app for playback at all now.  I use it only to download from my server and then I playback with the always reliable Smart Audiobook Player

I don't expose my server to the internet.  I use tailscale so that I can download from the server when I'm away from home.

In summary:  I love the project and the server is fantastic.  The functionality of the app is great. It has all of the features needed, but is not reliable enough to use for actually listening to books.

**Reply by sandreas** [reply]
>  I use it only to download from my server and then I playback with the always reliable Smart Audiobook Player

Interesting idea... maybe this is the way to go as long as the app is crashing. Thanks.

I'm working on my own app, but this is far from being production ready.

**Comment by politas** [comment]
I've just started using ABS, and I'm wanting to use it for podcasts to move away from Pocket Casts. I love the Pocket Casts user experience, but I don't like handing over my login credentials to them.

Out of the gate, I'm stunned that a podcast playlist doesn't operate as a queue in the Android client when the playlist says it is playing. One of my primary use cases is listening while driving, so I can't interact with my phone immediately (that's a legal requirement, as well as a safety issue).
